tools/nl/embeddings/build_embeddings_v2.py

```diff
@@ -0,0 +1,286 @@
+                    'Worksheet name in the Google Sheets file')
+
+flags.DEFINE_string('palm_alternatives_filename', 'palm_alternatives',
+                    'PaLM alternative descriptions csv filename')
+flags.DEFINE_string('other_alternatives_filename', 'other_alternatives',
+                    'Other alternative descriptions csv filename')
+
+flags.DEFINE_string('merged_output_filename_prefix', 'embeddings',
+                    'Final merged/concatenated SVs and descriptions file.')
+
+# Col names in the input files/sheets.
+SHEETS_DCID_COL = 'Id'
+SHEETS_NAME_COL = 'Name'
+SHEETS_DESCRIPTION_COL = 'Description'
+SHEETS_ALTERNATIVES_COL = 'Curated_Alternatives'
+SHEETS_OVERRIDE_COL = 'Override_Alternatives'
+
+CSV_DCID_COL = 'Id'
+CSV_PALM_ALTERNATIVES_COL = 'PaLM_Generated_Alternatives'
+CSV_OTHER_ALTERNATIVES_COL = 'Other_Alternatives'
+
+# Col names in the concatenated dataframe.
+COL_ID = 'dcid'
+COL_ALTERNATIVES = 'sentence'
+
+# Setting to a very high number right for now.
+MAX_ALTERNATIVES_LIMIT = 50
+
+
+def _add_sv(name: str, sv: str, text2sv: Dict[str, List[str]]) -> None:
+  if not name:
+    return
+  svs_list = []
+  if name in text2sv:
+    svs_list = text2sv[name]
+
+  if sv not in svs_list:
+    svs_list.append(sv)
+
+  text2sv[name] = svs_list
+
+
+def _get_texts_dcids(df: pd.DataFrame) -> Tuple[List[str], List[str]]:
+  text2sv_dict = {}
+  for _, row in df.iterrows():
+    sv = row[COL_ID].strip()
+    if COL_ALTERNATIVES in row:
+      alternatives = row[COL_ALTERNATIVES].split(';')
+
+      for alt in alternatives:
+        alt = alt.strip()
+        _add_sv(alt, sv, text2sv_dict)
+
+  texts = sorted(list(text2sv_dict.keys()))
+  dcids = [','.join(text2sv_dict[k]) for k in texts]
+
+  return (texts, dcids)
+
+
+def _trim_columns(df: pd.DataFrame) -> pd.DataFrame:
+  cols = [COL_ID, COL_ALTERNATIVES]
+  return df[cols]
+
+
+def _two_digits(number: int) -> str:
+  number_str = str(number)
+  if len(number_str) == 1:
+    number_str = "0" + number_str
+  return number_str
+
+
+def _make_gcs_embeddings_filename(filename_prefix: str) -> str:
+  now = datetime.datetime.now()
+
+  month_str = _two_digits(now.month)
+  day_str = _two_digits(now.day)
+  hour_str = _two_digits(now.hour)
+  minute_str = _two_digits(now.minute)
+  second_str = _two_digits(now.second)
+
+  return f"{filename_prefix}_curated_merged_alternatives_{now.year}_{month_str}_{day_str}_{hour_str}_{minute_str}_{second_str}.csv"
+
+
+def _merge_dataframes(df_1: pd.DataFrame, df_2: pd.DataFrame) -> pd.DataFrame:
+  return df_1.merge(df_2, how='left', on=COL_ID).fillna("")
+
+
+def _concat_alternatives(alternatives: List[str],
+                         max_alternatives,
+                         delimiter=";") -> str:
+  alts = set(alternatives[0:max_alternatives])
+  return f"{delimiter}".join(alts)
+
+
+def _split_alt_string(alt_string: str) -> List[str]:
+  alts = []
+  for alt in alt_string.strip().split(";"):
+    if alt and alt != ";":
```

Comment:
can alt have `;` given we're splitting by it?

Reply:
good point. removed